facade/service.go

```diff
@@ -1503,14 +1454,39 @@ func (f *Facade) ListTenants(ctx datastore.Context) ([]string, error) {
 	return tenantIDs, nil
 }
 
+// WaitServiceWithCancel waits for service/s to reach a particular desired state, or until canceled
+func (f *Facade) WaitSingleService(svc *service.Service, dstate service.DesiredState, cancel <-chan interface{}) error {
+	// error out if the desired state is invalid
+	if dstate.String() == "unknown" {
+		return fmt.Errorf("desired state unknown")
+	}
+
+	return f.zzk.WaitService(svc, dstate, cancel)
+
+}
+
 func (f *Facade) StartService(ctx datastore.Context, request dao.ScheduleServiceRequest) (int, error) {
 	defer ctx.Metrics().Stop(ctx.Metrics().Start("Facade.StartService"))
 	return f.ScheduleService(ctx, request.ServiceID, request.AutoLaunch, request.Synchronous, service.SVCRun)
 }
 
 func (f *Facade) RestartService(ctx datastore.Context, request dao.ScheduleServiceRequest) (int, error) {
 	defer ctx.Metrics().Stop(ctx.Metrics().Start("Facade.RestartService"))
-	return f.ScheduleService(ctx, request.ServiceID, request.AutoLaunch, request.Synchronous, service.SVCRestart)
+	forceRestart := func() (int, error) {
+		count, err := f.ScheduleService(ctx, request.ServiceID, request.AutoLaunch, true, service.SVCStop)
+		if err != nil {
+			return count, err
+		}
+
+		return f.ScheduleService(ctx, request.ServiceID, request.AutoLaunch, request.Synchronous, service.SVCRun)
+	}
+
+	if request.Synchronous {
+		return forceRestart()
+	} else {
+		go forceRestart()
+		return 0, nil
+	}
```

Comment:
Note: This is a "forcedRestart" (synchronous stop, then a start) in order to balance services while observing startLevel, which should be a separate function (or removed?) when rolling restarts is implemented.